cmaes/_mapcma.py

```diff
@@ -0,0 +1,429 @@
+from __future__ import annotations
+
+import math
+import numpy as np
+
+from typing import Any
+from typing import cast
+from typing import Optional
+
+
+_EPS = 1e-8
+_MEAN_MAX = 1e32
+_SIGMA_MAX = 1e32
+
+
+class MAPCMA:
+    """MAP-CMA stochastic optimizer class with ask-and-tell interface.
```

Comment:
It is helpful to mention the difference from the CMA-ES to understand MAP-CMA.
Please add an explanation like "The only difference from the CMA-ES is the mean vector update..."